packages/worker/src/handlers.ts

```diff
@@ -1,83 +1,105 @@
  *
  * This utility allows you to define handlers outside of the worker creation,
  * providing better code organization and reusability.
  *
+ * Supports three patterns:
+ * 1. Simple handler: just the function
+ * 2. Handler with prefetch: [handler, { prefetch: 10 }]
+ * 3. Batch handler: [batchHandler, { batchSize: 5, batchTimeout: 1000 }]
+ *
  * @template TContract - The contract definition type
  * @template TName - The consumer name from the contract
  * @param contract - The contract definition containing the consumer
  * @param consumerName - The name of the consumer from the contract
  * @param handler - The async handler function that processes messages
+ * @param options - Optional consumer options (prefetch, batchSize, batchTimeout)
  * @returns A type-safe handler that can be used with TypedAmqpWorker
  *
  * @example
  * ```typescript
  * import { defineHandler } from '@amqp-contract/worker';
  * import { orderContract } from './contract';
  *
- * // Define handler outside of worker creation
+ * // Simple handler without options
  * const processOrderHandler = defineHandler(
  *   orderContract,
  *   'processOrder',
  *   async (message) => {
- *     // message is fully typed based on the contract
  *     console.log('Processing order:', message.orderId);
  *     await processPayment(message);
  *   }
  * );
  *
- * // Use the handler in worker
- * const worker = await TypedAmqpWorker.create({
- *   contract: orderContract,
- *   handlers: {
- *     processOrder: processOrderHandler,
- *   },
- *   connection: 'amqp://localhost',
- * });
- * ```
- *
- * @example
- * ```typescript
- * // Define multiple handlers
- * const processOrderHandler = defineHandler(
+ * // Handler with prefetch
+ * const processOrderWithPrefetch = defineHandler(
  *   orderContract,
  *   'processOrder',
  *   async (message) => {
  *     await processOrder(message);
- *   }
+ *   },
+ *   { prefetch: 10 }
  * );
  *
- * const notifyOrderHandler = defineHandler(
+ * // Batch handler
+ * const processBatchOrders = defineHandler(
  *   orderContract,
- *   'notifyOrder',
- *   async (message) => {
- *     await sendNotification(message);
- *   }
- * );
- *
- * // Compose handlers
- * const worker = await TypedAmqpWorker.create({
- *   contract: orderContract,
- *   handlers: {
- *     processOrder: processOrderHandler,
- *     notifyOrder: notifyOrderHandler,
+ *   'processOrders',
+ *   async (messages) => {
+ *     await db.insertMany(messages);
  *   },
- *   connection: 'amqp://localhost',
- * });
+ *   { batchSize: 5, batchTimeout: 1000 }
+ * );
  * ```
  */
+export function defineHandler<
+  TContract extends ContractDefinition,
+  TName extends InferConsumerNames<TContract>,
+>(
+  contract: TContract,
+  consumerName: TName,
+  handler:
+    | WorkerInferConsumerHandler<TContract, TName>
+    | WorkerInferConsumerBatchHandler<TContract, TName>,
+): WorkerInferConsumerHandlerEntry<TContract, TName>;
```

Comment:
```suggestion
export function defineHandler<
  TContract extends ContractDefinition,
  TName extends InferConsumerNames<TContract>,
>(
  contract: TContract,
  consumerName: TName,
  handler: WorkerInferConsumerHandler<TContract, TName>,
): WorkerInferConsumerHandlerEntry<TContract, TName>;
```

we cannot have batch handling without batch config, it must be the rule and it must be documented as it